### PR TITLE
Domestic applicant registration

### PR DESCRIPTION
**Pull request by davidklaver**

### CODE REVIEW

File: app/controllers/charges_controller.rb [hunk 1]
```diff
@@ -4,11 +4,15 @@ def new
   end
 
   def create
+    
+    # amount = (current_user.charges.last.amount * 100).to_i
```
Comment (trevordjones) [inline app/controllers/charges_controller.rb]: Get rid of this line

File: app/controllers/charges_controller.rb [hunk 2]
```diff
@@ -4,11 +4,15 @@ def new
...
+    course_name = current_user.course_registrations.last.course.name
+    amount = current_user.course_registrations.last.course.price.to_i * 100
+
     p "*" * 50
-    p "hello"
+    p ":" + course_name
+    p amount.to_s
```
Comment (trevordjones) [inline app/controllers/charges_controller.rb]: Remove print statements

File: app/controllers/charges_controller.rb [hunk 3]
```diff
@@ -17,19 +21,19 @@ def create
...
 
   rescue Stripe::CardError => e
     flash[:error] = e.message
     redirect_to new_charge_path
   end
-  
+
   def show
     @charge = Charge.find(params[:id])
   end
```
Comment (trevordjones) [inline app/controllers/charges_controller.rb]: Undo all changes made to this controller

File: app/controllers/course_registrations_controller.rb [hunk 4]
```diff
@@ -2,16 +2,33 @@ class CourseRegistrationsController < ApplicationController
 
 	def new
 		@courses = Course.all
-		@course_registration = CourseRegistration.new
```
Comment (trevordjones) [inline app/controllers/course_registrations_controller.rb]: Fix up text editor - we want our indentation to be consistent

File: app/controllers/course_registrations_controller.rb [hunk 5]
```diff
@@ -2,16 +2,33 @@ class CourseRegistrationsController < ApplicationController
...
+    charge = Stripe::Charge.create(
+      :customer    => customer.id,
+      :amount      => amount,
+      :description => course.name,
+      :currency    => 'usd'
+    )
+
+    @new_charge = current_user.charges.create(uid: charge.id, amount: charge.amount / 100, description: charge.description, customer_id: customer.id)
+    current_user.update(course_id: course.id)
+    render "show.html.erb"
```
Comment (trevordjones) [inline app/controllers/course_registrations_controller.rb]: `render 'show'` or `render :show`

File: app/controllers/course_registrations_controller.rb [hunk 6]
```diff
@@ -2,16 +2,33 @@ class CourseRegistrationsController < ApplicationController
...
+
+  rescue Stripe::CardError => e
+    flash[:error] = e.message
+    redirect_to "/course_registrations/new"
 	end
 
 	def course_registration_params
```
Comment (trevordjones) [inline app/controllers/course_registrations_controller.rb]: Change to `course_params`
Also change method names that use `course_registration_params`

File: app/controllers/courses_controller.rb [hunk 7]
```diff
@@ -4,11 +4,15 @@ def index
 		@courses = Course.all
 	end
 
+	def show
+		@course = Course.find(params[:id])
```
Comment (trevordjones) [inline app/controllers/courses_controller.rb]: indentation

File: app/controllers/courses_controller.rb [hunk 8]
```diff
@@ -4,11 +4,15 @@ def index
...
+	end
+
 	def new
-		
+
```
Comment (trevordjones) [inline app/controllers/courses_controller.rb]: Get rid of methods we are not using

File: app/views/course_registrations/new.html.erb [hunk 9]
```diff
@@ -1,6 +1,31 @@
+<article>
+  <% if flash[:error].present? %>
+    <div id="error_explanation">
```
Comment (trevordjones) [inline app/views/course_registrations/new.html.erb]: Use bootstrap alerts here
```html
<div class="alert alert-danger">
</div>
```

File: app/views/course_registrations/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+<h1>Congrats! You are officially enrolled! </h1>
+<ul> 
+	<li> Amount: <%= number_to_currency(@new_charge.amount) %></li>
+	<li> Description: <%= @new_charge.description %></li>
+	<li> Email: <%= @new_charge.user.email %></li>
+	<li> Reference Number: <%= @new_charge.uid %></li>
```
Comment (trevordjones) [inline app/views/course_registrations/show.html.erb]: fix indentations here

File: app/views/courses/index.html.erb [hunk 11]
```diff
@@ -1,3 +1,3 @@
 <% @courses.each do |course| %>
-	<p><%= course.name %></p>
-<% end %>
\ No newline at end of file
+	<p><a href="/courses/<%= course.id %>"><%= course.name %></a></p>
```
Comment (trevordjones) [inline app/views/courses/index.html.erb]: Try a `link_to` here

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -36,7 +36,8 @@
   <script src="https://cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv.min.js"></script>
   <script src="https://cdnjs.cloudflare.com/ajax/libs/respond.js/1.4.2/respond.min.js"></script>
   <![endif]-->
-
+  <script src="https://js.stripe.com/v2/"></script>
+  <script src="https://js.stripe.com/v3/"></script>
```
Comment (trevordjones) [inline app/views/layouts/application.html.erb]: Verify that we need both versions

File: db/schema.rb [hunk 13]
```diff
@@ -10,6 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
+
```
Comment (trevordjones) [inline db/schema.rb]: Get rid of this extra space

File: app/views/courses/index.html.erb [hunk 14]
```diff
@@ -1,3 +1,3 @@
 <% @courses.each do |course| %>
-	<p><%= course.name %></p>
-<% end %>
\ No newline at end of file
+	<p><%= link_to( course.name, "/courses/#{course.id}") %></p>
```
Comment (davidklaver) [inline app/views/courses/index.html.erb]: `link_to(course.name, course_path(course.id)`

**Pull request merged**